File: zhiyoufy-web/src/model/dto/pms/PmsJobTemplateParam.ts
export interface PmsJobTemplateParam {
  projectId: number;
  projectName: string;

  folderId: number;
  folderName: string;

  name: string;
  description: string;
  jobPath: string;
  workerLabels: string;
  timeoutSeconds: number;
  baseConfPath: string;
  privateConfPath: string;
  configSingles: string;
  configCollections: string;
  extraArgs: string;
  isPerf: boolean;
  dashboardAddr: string;
}

File: zhiyoufy-server-java/zhiyoufy-server/src/main/java/com/example/zhiyoufy/server/domain/dto/pms/PmsJobTemplateParam.java
package com.example.zhiyoufy.server.domain.dto.pms;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class PmsJobTemplateParam {
	@NotNull
	private Long projectId;
	@NotEmpty
	private String projectName;

	@NotNull
	private Long folderId;
	@NotEmpty
	private String folderName;

	@NotEmpty
	@Size(min = 3)
	private String name;

	private String description;

	@NotNull
	private String jobPath;

	private String workerLabels;

	@NotNull
	private Integer timeoutSeconds;

	private String baseConfPath;

	private String privateConfPath;

	private String configSingles;
	private String configCollections;

	private String extraArgs;

	private Boolean isPerf;
	private String dashboardAddr;
}
